### ISSUE TITLE
jsx-no-bind should detect for ::

### ISSUE THREAD

**Issue by ha404** [yannickcr/eslint-plugin-react#795]
Hi! I love this rule: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md

The only fault is it's not capturing `::` which essentially is `bind(this)`.

**Comment by ljharb** [comment]
`::` by itself isn't valid - it definitely should be catching `this::foo` or `::foo` or `a::b.c` etc though. Can you provide specific code that it's not catching?

**Comment by hawsome** [comment]
Oh yeah, it's like:
```jsx
render () {
  <div onClick={::this.handleClick} />
}
```

**Comment by ljharb** [comment]
Fixed by #644.
